src/mainwindow.cpp

```diff
@@ -48,8 +46,9 @@ void MainWindow::Extract()
 
 void MainWindow::Compress()
 {
-    compressWindow->show();
-    this->hide();
+    CompressWindow *compressWindow = new CompressWindow(this);
```

Comment:
I think this will make memory leak, can you prove this safe?

Reply:
I think, is it better to reuse kind? Or something like this? https://www.qtcentre.org/threads/31185-How-to-correctly-delete-a-QDialog-created-via-new?s=648484ce83aae176e610f73ab3588e79&p=145569#post145569